packages/angular-material/src/other/table.renderer.ts

```diff
@@ -0,0 +1,154 @@
+*/
+import startCase from 'lodash/startCase';
+import { NgRedux } from '@angular-redux/store';
+import { Component } from '@angular/core';
+import { JsonFormsArrayControl } from '@jsonforms/angular';
+import {
+  ArrayControlProps,
+  ControlElement,
+  isObjectArrayControl,
+  isPrimitiveArrayControl,
+  JsonFormsState,
+  JsonSchema,
+  or,
+  OwnPropsOfRenderer,
+  // Paths,
+  Paths,
+  RankedTester,
+  rankWith,
+  UISchemaElement
+} from '@jsonforms/core';
+
+@Component({
+  selector: 'ObjectRenderer',
+  template: `
+    <table
+      mat-table
+      [dataSource]="data"
+      class="mat-elevation-z8"
+      [trackBy]="trackElement"
+    >
+      <ng-container
+        *ngFor="let item of items"
+        matColumnDef="{{ item.property }}"
+      >
+        <th mat-header-cell *matHeaderCellDef>{{ item.header }}</th>
+        <td mat-cell *matCellDef="let index = index">
+          <!--{{ element[item.property] }}-->
+          <jsonforms-outlet
+            [renderProps]="getProps(index, item.props)"
+          ></jsonforms-outlet>
+        </td>
+      </ng-container>
+
+      <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+      <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+    </table>
+  `,
+  styles: ['table {width: 100%;}']
+})
+export class TableRenderer extends JsonFormsArrayControl {
+  detailUiSchema: UISchemaElement;
+  displayedColumns: string[];
+  items: ColumnDescription[];
+
+  constructor(ngRedux: NgRedux<JsonFormsState>) {
+    super(ngRedux);
+  }
+  trackElement(index: number, _element: any) {
+    return index ? index : null;
+  }
+  mapAdditionalProps(props: ArrayControlProps) {
+    this.items = this.generateCells(props.schema, props.path);
+    this.displayedColumns = this.items.map(item => item.property);
+  }
+  getProps(index: number, props: OwnPropsOfRenderer): OwnPropsOfRenderer {
+    const rowPath = Paths.compose(
+      props.path,
+      `${index}`
+    );
+    return {
+      schema: props.schema,
+      uischema: props.uischema,
+      path: rowPath
+    };
+  }
+  generateCells = (
+    schema: JsonSchema,
+    rowPath: string
+  ): ColumnDescription[] => {
+    if (schema.type === 'object') {
+      return this.getValidColumnProps(schema).map(prop => {
+        return {
+          property: prop,
+          header: startCase(prop),
+          props: {
+            schema: schema,
+            uischema: controlWithoutLabel(`#/properties/${prop}`),
+            path: rowPath
+          }
+        };
+      });
+    }
+    return [
+      {
+        property: 'DUMMY',
+        header: this.label,
+        props: {
+          schema: schema,
+          uischema: controlWithoutLabel(`#`),
+          path: rowPath
+        }
+      }
+    ];
+  };
+
+  getValidColumnProps = (scopedSchema: JsonSchema) => {
+    if (scopedSchema.type === 'object') {
+      return Object.keys(scopedSchema.properties).filter(
+        prop => scopedSchema.properties[prop].type !== 'array'
```

Comment:
Do we only care about arrays being filtered out? Shouldn't we filter objects as well?

Reply:
I copied from the react renderer. Yes it would make sense to filter out objects, too.